wayland-scanner/src/lib.rs

```diff
@@ -49,10 +47,197 @@ mod protocol;
 mod server_gen;
 mod util;
 
+// `bytes`, `next_chr`, `parse_lit_str`, `parse_lit_str_cooked` and `parse_lit_str_raw` are adapted
+// from syn:
+// https://github.com/dtolnay/syn/blob/362ee2d02df3f1b2e74c7b7a4cf2ed3c106404c9/src/lit.rs#L1062-L1167
+// and
+// https://github.com/dtolnay/syn/blob/362ee2d02df3f1b2e74c7b7a4cf2ed3c106404c9/src/lit.rs#L1327-L1388
```

Comment:
Could you instead put all this parsing code into its own file, to avoid overloading hte `lib.rs` file?

Reply:
Moved